### ISSUE TITLE
Unable to Authorize with Plotly Cloud

### ISSUE THREAD

**Issue by snooc** [plotly/falcon-sql-client#227]
When running falcon in the docker container, the OAuth callback fails with "Error: Please login to access this page."

Relevant logs:

```
{"name":"plotly-database-connector-logger","hostname":"0f44c726d23e","pid":24,"level":30,"msg":"Request: /oauth2/callback","time":"2017-11-09T13:26:31.816Z","v":0}
{"name":"plotly-database-connector-logger","hostname":"0f44c726d23e","pid":24,"level":30,"msg":"Request: /static/web-bundle.min.js?t=1510233991030","time":"2017-11-09T13:26:31.878Z","v":0}
{"name":"plotly-database-connector-logger","hostname":"0f44c726d23e","pid":24,"level":30,"msg":"Request: /oauth2/oauth2","time":"2017-11-09T13:26:32.784Z","v":0}
```

Screenshot of error:

![screen shot 2017-11-09 at 8 23 50 am](https://user-images.githubusercontent.com/547505/32607612-9aedf6dc-c527-11e7-8409-45e3097ed0eb.png)

**Comment by tarzzz** [comment]
This shall be fixed in PR: #223 
specifically [this commit](https://github.com/plotly/falcon-sql-client/pull/223/commits/1dfc85cdd5b5a22a4d9fc992919fd18f513b3c5b)

**Comment by n-riesco** [comment]
@tarzzz One question: can we fix the issue here just by pushing the commit https://github.com/plotly/falcon-sql-client/pull/223/commits/1dfc85cdd5b5a22a4d9fc992919fd18f513b3c5b 
 into master? or we actually need the whole #223 ?

If https://github.com/plotly/falcon-sql-client/pull/223/commits/1dfc85cdd5b5a22a4d9fc992919fd18f513b3c5b fixes the issue here, why don't we open a PR only with this commit and merge it into master straight away? In this way, we could take on @snooc 's offer to create a PR for #224.

**Comment by tarzzz** [comment]
Yes, I am going to push a separate PR with just the commit(s) required to
fix the issue today.

On Friday, November 10, 2017, Nicolas Riesco <notifications@github.com>
wrote:
> @tarzzz One question: can we fix the issue here just by pushing the
commit 1dfc85c
> into master? or we actually need the whole #223 ?
>
> If 1dfc85c fixes the issue here, why don't we open a PR only with this
commit and merge it into master straight away? In this way, we could take
on @snooc 's offer to create a PR for #224.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.<
https://ci3.googleusercontent.com/proxy/cLkw91GwvhzjswWJg19HUNBMJAfC1FMoesSxx0KPObAdXAnhgBoOfAqdrEU0387GmT4HqyEkAfzaY-_kzZVhaxBKJd_3yOojmNEVCFmpMxD_m2iizqnfLUX1I6ipHpGC6a0osKcWnvFJx38HSkS3bPRPAWvFbw=s0-d-e1-ft#https://github.com/notifications/beacon/ABeZJCNY9ZQqNkAYQJLygxjYx3x-5jnXks5s1ALkgaJpZM4QX9zX.gif>

**Comment by snooc** [comment]
Using @tarzzz's fix-authorization branch, I am now seeing a different error:

`Error: User myusername is not allowed to view this app`

**Comment by tarzzz** [comment]
@snooc: Yes, that is a valid error. The server version only allows whitelisted users to authenticate to db connector (For security reasons).

You need to add the following lines to your `settings.yaml`:

```
ALLOWED_USERS:
    - myUsername1
    - AnyOtherUserYouWantToGiveAccess
```

Alternatively, you can set an environment variable for your docker container containing whitelisted users:
`PLOTLY_CONNECTOR_ALLOWED_USERS` with values as: `yourUsername,YourOtherUsername`



For development purposes, you can also disable authentication by either setting:

```
AUTH_ENABLED: true
```
in your `settings.yaml` OR as the environment variable:

```
PLOTLY_CONNECTOR_AUTH_ENABLED: "true"
```

Note: Disabling Authentication is **not** recommended for production purposes.

**Comment by snooc** [comment]
Thanks.

After setting `PLOTLY_CONNECTOR_ALLOWED_USERS=myusername` I am able to get past the authorization process, but after authorization succeeds, localhost:9494 is always the login screen. Are there any other configuration options that I may be missing?

I am using the following command to start the container. The container that I am user is build using the fix-authentication branch.

```docker run -it --rm -d -p 9494:9494 -e PLOTLY_CONNECTOR_ALLOWED_USERS=myusername falcon```

Also, setting `PLOTLY_CONNECTOR_AUTH_ENABLES=false` did work for me, but obviously it doesn't confirm this issue.

**Comment by tarzzz** [comment]
Can you please  try:

  - Refreshing the `localhost:9494` to see if you see the connections.
  - Send a screenshot of cookies being set (Using a browser cookie manager) on `localhost` after authentication part is successful? 

Ah one more thing. since you are running it locally, you also needs to set env variable: `PLOTLY_CONNECTOR_SSL_ENABLED=false` in your docker run command -- because by default it expects SSL certificates (again a security measure)..

**Comment by n-riesco** [comment]
@tarzzz I think what @snooc is seeing is caused by the RegExp issue you told me on slack.

@snooc If possible, could you do `git checkout 3201a0dad9fe1f81b269259cb8b247bfaf72f4b4`? This should land you on a commit before the issue with RegExp was introduced.

**Comment by snooc** [comment]
Setting `PLOTLY_CONNECTOR_SSL_ENABLED=false` fixed the issue that I was seeing.

Should I be able to connect to Plot.ly? The generating certificates stages seems to be failing or taking a lot longer than I expect. It's been about 20+ minutes. I see no logs reporting a failure.

**Comment by n-riesco** [comment]
> The generating certificates stages seems to be failing or taking a lot longer than I expect.

Yes, in my experience they can take a long time, but once generated they are saved so that future sessions don't need to generate them again (unless they expire).

>  I see no logs reporting a failure.

Logs are stored in `~/.plotly/connector/log.log`.

I've had a quick look through the code and if I understood correctly Falcon:
- tries 5 times to create a certificate
- if unable to generate the certificate, then an error is logged in `log.log`
- but nothing is reported to the user

So I'd recommend to restart falcon if after 5-6 min the certificates haven't been generated.

I've opened issue #237 to address this.

**Comment by tarzzz** [comment]
> Should I be able to connect to Plot.ly? 
Yes.. you dont require certificates for server version. It is only meant for standalone electron apps.

**Comment by n-riesco** [comment]
Closed via #236 and #253
